plugins/single_plugins/wsets.cpp

```diff
@@ -113,6 +121,48 @@ class wayfire_wsets_plugin_t : public wf::plugin_interface_t
     std::list<wf::activator_callback> send_callback;
     std::map<int, std::shared_ptr<wf::workspace_set_t>> available_sets;
 
+    wf::ipc::method_callback attach_to_output = [=] (nlohmann::json data)
+    {
+        WFJSON_EXPECT_FIELD(data, "id", number_integer);
+        WFJSON_EXPECT_FIELD(data, "output", number_integer);
+        auto wset = wf::ipc::find_workspace_set_by_id(data["id"]);
+        if (!wset)
+        {
+            return wf::ipc::json_error("workspace set not found");
+        }
+
+        auto o = wf::ipc::find_output_by_id(data["output"]);
+        if (!o)
+        {
+            return wf::ipc::json_error("output not found");
+        }
+
+        auto response = wf::ipc::json_ok();
+        wset->attach_to_output(o);
+        return response;
+    };
+
+    wf::ipc::method_callback set_output_wset = [=] (nlohmann::json data)
+    {
+        WFJSON_EXPECT_FIELD(data, "id", number_integer);
+        WFJSON_EXPECT_FIELD(data, "wset", number_integer);
+        auto o = wf::ipc::find_output_by_id(data["id"]);
+        if (!o)
+        {
+            return wf::ipc::json_error("output not found");
+        }
+
+        auto wset = wf::ipc::find_workspace_set_by_id(data["wset"]);
+        if (!wset)
+        {
+            return wf::ipc::json_error("workspace set not found");
+        }
+
+        auto response = wf::ipc::json_ok();
+        o->set_workspace_set(wset->shared_from_this());
```

Comment:
This should call `select_workspace()` if we use index instead of id - the function might need to be refactored a bit so that it gets the output as an additional parameter, which defaults to the currently focused output (but can also be set to an arbitrary output)